src/views/index-js.mustache

```diff
@@ -2691,12 +2695,12 @@ function setIndividualPokemonHidden (id) {
 }
 
 // eslint-disable-next-line no-unused-vars
-function addPokemonFilter (pokemonId, show) {
-    pokemonFilter[pokemonId].show = show;
+function addPokemonFilter (pokemonIdString, pokemonId, formId, show) {
+    pokemonFilter[pokemonIdString].show = show;
     store('pokemon_filter', JSON.stringify(pokemonFilter));
 
     $.each(pokemonMarkers, function (index, pokemon) {
-        if (pokemon.pokemon_id === pokemonId) {
+        if (pokemon.pokemon_id === pokemonId && pokemon.form_id === formId) {
```

Comment:
```suggestion
        if (pokemon.pokemon_id === pokemonId && pokemon.form === formId) {
```

Reply:
Indeed. `form_id` was used for `quest_rewards`.